File: 01_selective_logging/05_pos_processing.js

var asset = 'projects/ee-simex/assets/classification';

var assetUf = 'projects/mapbiomas-workspace/AUXILIAR/estados-2016'

var assetLulc = 'projects/mapbiomas-workspace/public/collection8/mapbiomas_collection80_integration_v1'







var roi = ee.FeatureCollection(assetUf).filter('NM_ESTADO == "PARأپ"')


var lulc = ee.Image(assetLulc).select('classification_2022').clip(roi.geometry());

var water = lulc.eq(33).distance(ee.Kernel.euclidean(5)).gt(0);

var forest = lulc.eq(3);


var freq = ee.ImageCollection(asset)
    .map(function(img){return img.gt(50)}).sum()
    .clip(roi.geometry())
    .mask(forest.eq(1)).where(water.eq(1), 0);
    



var vis = {
    'min': 0,
    'max': 16,
    'palette': [
      "#000000",
      "ffffcc",
      "ffeda0",
      "fed976",
      "feb24c",
      "fd8d3c",
      "fc4e2a",
      "e31a1c",
      "bd0026",
      "800026"
    ],
    //'format': 'png'
};

Map.addLayer(freq, vis, 'freq');

File: 01_selective_logging/05_pos_processing.py
import ee

# PROJECT = 'ee-mapbiomas-imazon'
PROJECT = 'ee-simex'

ee.Authenticate()
ee.Initialize(project=PROJECT)

asset = 'projects/ee-simex/assets/logging_predictions'
assetOutput = 'projects/ee-simex/assets/predictions'

mb = ee.Image('projects/mapbiomas-workspace/public/collection8/mapbiomas_collection80_integration_v1')
mb = mb.select('classification_2020')
mb = ee.Image(mb.eq(3).Or(mb.eq(6)))

collection = ee.ImageCollection(asset)
collectionLoaded = ee.ImageCollection(assetOutput)
image = collection.reduce(ee.Reducer.sum())

idsLoaded = collectionLoaded.reduceColumns(ee.Reducer.toList(), ['image_id']).get('list').getInfo()
ids = collection.filter(ee.Filter.inList('image_id', idsLoaded).Not()) \
    .reduceColumns(ee.Reducer.toList(), ['image_id']) \
    .get('list').getInfo()

high = image.gt(3)
kernel = ee.Kernel.square(radius=10)
buffer = high.focalMax(kernel=kernel, iterations=2).reproject(crs='EPSG:4326', scale=30)
binaryMask = buffer.unmask(0)

for id in ids:
    image = ee.Image(collection.filter(ee.Filter.eq('image_id', id)).first())
    fixImage = ee.Image(image.where(binaryMask.eq(0), 0))
    fixImage = ee.Image(fixImage.mask(mb.eq(1)).selfMask())
    fixImage = ee.Image(fixImage.copyProperties(image))
    fixImage = fixImage.set('system:time_end', image.get('system:time_end'))
    fixImage = fixImage.set('system:time_start', image.get('system:time_start'))

    print(f'processing {id}')

    task = ee.batch.Export.image.toAsset(
        image=fixImage,
        description=id,
        assetId=assetOutput + '/' + id.replace('.tif','') + '_1',
        pyramidingPolicy={'.default': 'mode'},
        region=image.geometry(),
        scale=10,
        maxPixels=1e13
    )
    task.start()
